promise.go

```diff
@@ -200,41 +200,47 @@ func (promise *Promise) Await() (interface{}, error) {
 // from the resolved promises in the same order as defined in the iterable of multiple promises.
 // If it rejects, it is rejected with the reason from the first promise in the iterable that was rejected.
 func All(promises ...*Promise) *Promise {
-	var length = len(promises)
+	psLen := len(promises)
 
-	if promises == nil || length == 0 {
+	if psLen == 0 {
 		return Resolve(make([]interface{}, 0))
 	}
 
 	return New(func(resolve func(interface{}), reject func(error)) {
-		var resolutionsChan = make(chan []interface{}, length)
-		var errorChan = make(chan error, length)
+		resolutionsChan := make(chan []interface{}, psLen)
+		errorChan := make(chan error, psLen)
 
-		for index, promise := range promises {
-			func(x int) {
-				promise.Then(func(data interface{}) interface{} {
-					resolutionsChan <- []interface{}{x, data}
+		for ind, promise := range promises {
+			func (i int, p *Promise) {
+				p.Then(func(data interface{}) interface{} {
+					resolutionsChan <- []interface{}{i, data}
 					return data
-				})
-				promise.Catch(func(err error) error {
+				}).Catch(func(err error) error {
 					errorChan <- err
 					return err
 				})
-			}(index)
+			}(
+				ind,
+				promise,
+			)
 		}
 
-		var resolutions = make([]interface{}, length)
-		for x := 0; x < length; x++ {
+		resolutions := make([]interface{}, psLen)
+		numResolved := 0
+		for {
```

Comment:
Do you mind rewriting this "infinite" for loop to use
```go
for x := 0; x < psLen; x++ {}
```
Look clearer in my opinion and you don't have to manage numsResolved independently

Reply:
I hear ya though, isn't `select` non blocking?  Would mean that for loop may complete before channels are done piping there content, correct?

Reply:
Example:
https://tour.golang.org/concurrency/5

Reply:
Hmm
```
A select *blocks* until one of its cases can run, then it executes that case. It chooses one at random if multiple are ready.
```

Reply:
Woops *face palm*  I could've sworn it was non blocking (I forgot it is non blocking if you give it a way out (a non-blocking `default` case)lol).  Yeah you're right we could write it with the `for x...` loop,  (I'll update on my side),

Reply:
correction: `select` still is blocking regardless *face palm* (hahaha),

Reply:
No worries 😄